### PR TITLE
fix(quick filter): Selected option get lost if they are not within max items size

### PR DESCRIPTION
**Pull request by moertenschlag**
…oupitems (APM-266050)

### <strong>Pull Request</strong>

<hr>
Hi, thank you for contributing to Barista with this pull request (PR).

To ensure a fast process and merging of your PR please make sure it fulfills the
coding standards and contribution guidelines.

- A feature proposal has been provided, discussed and approved first.
- There is a meaningful description of the issue in GitHub (Screenshots are
  often helpful).
- If the PR introduces breaking-changes or deprecations it matches the following
  guidelines.
  - The commit message follows our commit guidelines.
  - Tests for the changes have been added (for bug fixes / features).
  - Docs have been added / updated (for bug fixes / features).

Please choose the type appropriate for the changes below: <br>

#### Checklist

- [x] I have read the CONTRIBUTING doc and I follow the PR guidelines
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: libs/barista-components/quick-filter/src/quick-filter-group.ts [hunk 1]
```diff
@@ -235,8 +235,12 @@ export class DtQuickFilterGroup<T = any> implements AfterViewInit {
       );
       if (!this.isDetail && items.length > this.maxGroupItems) {
         this._truncatedGroupItems = true;
-        this._showMoreCount = items.length - this.maxGroupItems;
-        return items.slice(0, this.maxGroupItems);
+        const filteredItems = items.filter(
+          (item, idx) => idx <= this.maxGroupItems || this._isActive(item),
```
Comment (tomheller) [inline libs/barista-components/quick-filter/src/quick-filter-group.ts:239]: Please do not abbreviate variable names where it is not entirely clear on what the variable should be.
See [Coding standards](https://github.com/dynatrace-oss/barista/blob/master/CODING_STANDARDS.md#naming)

```suggestion
          (item, index) => index <= this.maxGroupItems || this._isActive(item),
```

**Review by tomheller — APPROVED** [review]
Thank you for contributing this fix @moertenschlag!
one small nitpick comment added. 

Please also sign the CLA, as this is required for us to be able to accept your contribution.

**Review by thomaspink — APPROVED** [review]
LGTM. Thank you

**Comment by tomheller** [comment]
It looks like that there are some e2e tests that are still failing, as you are changing the expected behavior. You might need to adjust this as well.

**Review by tomheller — DISMISSED** [review]

**Comment by ffriedl89** [comment]
@moertenschlag Thx for the contribution, I've rebased and squashed your commits and we will merge the PR. You don't have to take any further action. The changeset will be included in the next release.

**Review by ffriedl89 — APPROVED** [review]

**Pull request merged**